modules/jolt_physics/objects/jolt_area_3d.cpp

```diff
@@ -0,0 +1,657 @@
+	ERR_FAIL_NULL(other_object);
+
+	p_overlap.rid = other_object->get_rid();
+	p_overlap.instance_id = other_object->get_instance_id();
+
+	ShapeIndexPair &shape_indices = p_overlap.shape_pairs[{ p_other_shape_id, p_self_shape_id }];
+
+	shape_indices.other = other_object->find_shape_index(p_other_shape_id);
+	shape_indices.self = find_shape_index(p_self_shape_id);
+
+	p_overlap.pending_added.push_back(shape_indices);
+}
+
+bool JoltArea3D::_remove_shape_pair(Overlap &p_overlap, const JPH::SubShapeID &p_other_shape_id, const JPH::SubShapeID &p_self_shape_id) {
+	HashMap<ShapeIDPair, ShapeIndexPair, ShapeIDPair>::Iterator shape_pair = p_overlap.shape_pairs.find(ShapeIDPair(p_other_shape_id, p_self_shape_id));
+
+	if (shape_pair == p_overlap.shape_pairs.end()) {
+		return false;
+	}
+
+	p_overlap.pending_removed.push_back(shape_pair->value);
+	p_overlap.shape_pairs.remove(shape_pair);
+
+	return true;
+}
+
+void JoltArea3D::_flush_events(OverlapsById &p_objects, const Callable &p_callback) {
+	for (OverlapsById::Iterator E = p_objects.begin(); E;) {
+		Overlap &overlap = E->value;
+
+		if (p_callback.is_valid()) {
+			for (ShapeIndexPair &shape_indices : overlap.pending_removed) {
+				_report_event(p_callback, PhysicsServer3D::AREA_BODY_REMOVED, overlap.rid, overlap.instance_id, shape_indices.other, shape_indices.self);
+			}
+
+			for (ShapeIndexPair &shape_indices : overlap.pending_added) {
+				_report_event(p_callback, PhysicsServer3D::AREA_BODY_ADDED, overlap.rid, overlap.instance_id, shape_indices.other, shape_indices.self);
+			}
+		}
+
+		overlap.pending_removed.clear();
+		overlap.pending_added.clear();
+
+		OverlapsById::Iterator next = E;
+		++next;
+
+		if (overlap.shape_pairs.is_empty()) {
+			p_objects.remove(E);
+		}
+
+		E = next;
+	}
+}
+
+void JoltArea3D::_report_event(const Callable &p_callback, PhysicsServer3D::AreaBodyStatus p_status, const RID &p_other_rid, ObjectID p_other_instance_id, int p_other_shape_index, int p_self_shape_index) const {
+	ERR_FAIL_COND(!p_callback.is_valid());
+
+	static thread_local Array arguments = []() {
+		Array array;
+		array.resize(5);
+		return array;
+	}();
+
+	arguments[0] = p_status;
+	arguments[1] = p_other_rid;
+	arguments[2] = p_other_instance_id;
+	arguments[3] = p_other_shape_index;
+	arguments[4] = p_self_shape_index;
+
+	p_callback.callv(arguments);
+}
+
+void JoltArea3D::_notify_body_entered(const JPH::BodyID &p_body_id) {
+	const JoltReadableBody3D jolt_body = space->read_body(p_body_id);
+
+	JoltBody3D *body = jolt_body.as_body();
+	if (unlikely(body == nullptr)) {
+		return;
+	}
+
+	body->add_area(this);
+}
+
+void JoltArea3D::_notify_body_exited(const JPH::BodyID &p_body_id) {
+	const JoltReadableBody3D jolt_body = space->read_body(p_body_id);
+
+	JoltBody3D *body = jolt_body.as_body();
+	if (unlikely(body == nullptr)) {
+		return;
+	}
+
+	body->remove_area(this);
+}
+
+void JoltArea3D::_force_bodies_entered() {
+	for (auto &[id, body] : bodies_by_id) {
```

Comment:
`auto` is [not allowed](https://docs.godotengine.org/en/stable/contributing/development/cpp_usage_guidelines.html#auto-keyword)